diff --git a/src/missingcites.cpp b/src/missingcites.cpp
index 235136e..b039eef 100644
--- a/src/missingcites.cpp
+++ b/src/missingcites.cpp
@@ -5,6 +5,7 @@
 // Copyright: See COPYING file that comes with this distribution
 //
 //
+#include <cstring>
 #include "missingcites.h"
 
 /**
diff --git a/src/namepoolfile.cpp b/src/namepoolfile.cpp
index f1579b0..7c9f149 100644
--- a/src/namepoolfile.cpp
+++ b/src/namepoolfile.cpp
@@ -4,6 +4,7 @@
 // Author: Michael Schmidt, Christoph Pinkel <>, (C) 2007-2008
 // Copyright: See COPYING file that comes with this distribution
 //
+#include <cstring>
 #include "namepoolfile.h"
 
 NamePoolFile* NamePoolFile::instance0 = 0L;
diff --git a/src/namepoolmgr.cpp b/src/namepoolmgr.cpp
index e7fef4b..438fe10 100644
--- a/src/namepoolmgr.cpp
+++ b/src/namepoolmgr.cpp
@@ -4,6 +4,7 @@
 // Author: Michael Schmidt, Christoph Pinkel <>, (C) 2007-2008
 // Copyright: See COPYING file that comes with this distribution
 //
+#include <cstring>
 #include "namepoolmgr.h"
 
 NamePoolMgr* NamePoolMgr::instance = 0L;
diff --git a/src/volumemgr.cpp b/src/volumemgr.cpp
index 3775797..753ddff 100644
--- a/src/volumemgr.cpp
+++ b/src/volumemgr.cpp
@@ -4,6 +4,7 @@
 // Author: Michael Schmidt, Christoph Pinkel <>, (C) 2007-2008
 // Copyright: See COPYING file that comes with this distribution
 //
+#include <stdlib.h>
 #include "volumemgr.h"
 
 VolumeMgr* VolumeMgr::instance = 0L;
